test/celeritas/track/TrackInit.test.cc

```diff
@@ -272,36 +255,32 @@ TEST_F(TrackInitTest, primaries)
     std::vector<char>      alive(num_tracks, 0);
     ITTestInput            input(alloc, alive);
 
-    for (auto i = num_primaries; i > 0; i -= capacity)
+    for (size_type i = 0; i < num_sets; ++i)
     {
-        EXPECT_EQ(track_init_states.num_primaries, i);
-
         // Create track initializers on device from primary particles
-        extend_from_primaries(track_inits->host_ref(), &track_init_states);
+        auto primaries = generate_primaries(num_primaries);
+        extend_from_primaries(core_data, primaries);
+        EXPECT_EQ(core_data.states.init.initializers.size(), num_primaries);
 
-        for (auto j = capacity; j > 0; j -= num_tracks)
+        for (size_type j = 0; j < primaries_per_track; ++j)
         {
-            EXPECT_EQ(track_init_states.initializers.size(), j);
-
             // Initialize tracks on device
-            initialize_tracks(core_data, &track_init_states);
+            initialize_tracks(core_data);
 
-            // Launch kernel that will kill all trackss
+            // Launch kernel that will kill all tracks
```

Comment:
It looks like maybe there's no test of adding some new primaries while others are still alive? If so, perhaps change the kernel to only kill half of the tracks?

Reply:
Agreed, will do.